lib/MusicBrainz/Server/DatabaseConnectionFactory.pm

```diff
@@ -67,8 +67,18 @@ sub get {
 
     my $database = $databases{$name};
 
-    if ($name eq 'MAINTENANCE' && !defined($database)) {
-        $database = $databases{READWRITE};
+    unless (defined $database) {
+        if ($name eq 'MAINTENANCE') {
+            $database = $databases{READWRITE};
+        } elsif ($name =~ /^SYSTEM_(.+)$/) {
+            my $database_name = $1;
+            my $system = $class->get('SYSTEM');
+            $database = $system->meta->clone_object(
+                $system,
+                database => $class->get($database_name)->database,
+            );
```

Comment:
It seems to pick the properties from the `SYSTEM` database connection definition but to overwrite `database` (name in PostgreSQL) with the one from the (expected) database connection definition to use. If so, it is indeed just doing the same as before.

I updated `admin/psql` help message accordingly in commit https://github.com/metabrainz/musicbrainz-server/pull/3079/commits/8b1565ddcea71eeb76e1e6a7cb5bf5240529912b.

Reply:
I would change "poweruser" to "superuser" since that is what the PostgreSQL documentation uses, but looks good otherwise. :+1: